src/server/openapi.yaml

```diff
@@ -0,0 +1,199 @@
+openapi: 3.1.0
+info:
+  title: CTB Backend API
+  version: 1.0.0
+servers:
+  - url: https://ctb-backend.onrender.com
+tags:
+  - name: auth
+    description: Authentication and user management
+  - name: stock
+    description: Stock market data and stock price prediction
+paths:
+  /api/v1/auth/register:
+    post:
+      tags:
+        - auth
+      summary: Register
+      description: Register a new user
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+        required: true
+      responses:
+        '201':
+          description: Successful operation
+        '202':
+          description: User with given name already exists
+        '400':
+          description: Invalid json format
+        '500':
+          description: Internal server error
+  /api/v1/auth/login:
+    post:
+      tags:
+        - auth
+      summary: Login
+      description: Login user.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+        required: true
+      responses:
+        '201':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Token'
+        '400':
+          description: Invalid json format
+        '401':
+          description: Unauthorized
+        '403':
+          description: Unauthorized - wrong password
+        '500':
+          description: Internal server error
+  /api/v1/auth/me:
+    get:
+      tags:
+        - auth
+      summary: Me
+      description: Get user's information
+      parameters:
+        - in: header
+          name: x-access-token
+          schema:
+            type: string
+          required: true
+          description: JWT access token
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Me'
+        '401':
+          description: Unauthorized
+        '500':
+          description: Internal server error
+      security:
+        - JWT: []
+  /api/v1/auth/logout:
+    post:
+      tags:
+        - auth
+      summary: Logout
+      description: Logout user
+      parameters:
+        - in: header
+          name: x-access-token
+          schema:
+            type: string
+          required: true
+          description: JWT access token
+      responses:
+        '201':
+          description: Successful operation
```

Comment:
This is not in scope of this PR but – I don't think returning 201 on logout is appropriate, it should probably be 205 ("The server successfully processed the request, asks that the requester reset its document view, and is not returning any content."). It's true that an expired token is inserted into database but that's an internal side-effect + we're not sending it back to the user. Similarly /login and /refresh should probably return 200 instead of 201.

Reply:
First part - we actually do send response (`{"message": "User is successfully logged out"}`) on successful  logout, so imho 200 would be more apprioprate.
Second part - yes, you are right, we should change it to 200.